packages/block-editor/src/components/block-lock/menu-item.js

```diff
@@ -13,8 +13,11 @@ import useBlockLock from './use-block-lock';
 import BlockLockModal from './modal';
 
 export default function BlockLockMenuItem( { clientId } ) {
-	const { canMove, canRemove, canLock } = useBlockLock( clientId, true );
-	const isLocked = ! canMove || ! canRemove;
+	const { canEdit, canMove, canRemove, canLock } = useBlockLock(
+		clientId,
+		true
+	);
+	const isLocked = ! canEdit || ! canMove || ! canRemove;
```

Comment:
Would it make sense to return `isLocked` from the `useBlockLock` hook? We are already using it in two places and if there are more complex or dynamic rules applied it would offload responsibility for managing this to a single source.

Reply:
we also use it as `isAllChecked` in another place

Reply:
and as inversion of isLocked `canEdit && canMove && canRemove`